lightpath/controller.py

```diff
@@ -484,28 +511,30 @@ def make_graph(
                 last_on_branch = i
 
             try:
-                next_dev = nodes[i+1][1]['dev']
+                next_dev = nodes[i+1][1]['res']
```

Comment:
Seems like with the `res`/`dev` stuff it might be worthwhile having a single keyed set of information:
```
@dataclass
class NodeMetadata:
    res: SearchResult
    dev: Optional[ophyd.Device]


node[...]["md"] = NodeMetadata(...)
```

Reply:
I like this